Replace PositiveIntegerOrAuto by ColumnCount

nox opened the pull request:
It was its only use.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/20104)
<!-- Reviewable:end -->

nox committed:
Replace PositiveIntegerOrAuto by ColumnCount

It was its only use.

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/properties/gecko.mako.rs, components/style/values/specified/column.rs, components/style/values/generics/column.rs, components/style/properties/data.py, components/style/values/specified/mod.rs and 5 more
 * @canaltinova: components/style/properties/gecko.mako.rs, components/style/values/specified/column.rs, components/style/values/generics/column.rs, components/style/properties/data.py, components/style/values/specified/mod.rs and 5 more
 * @emilio: components/style/properties/gecko.mako.rs, components/style/values/specified/column.rs, components/style/values/generics/column.rs, components/style/properties/data.py, components/style/values/specified/mod.rs and 6 more

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!
* These commits modify style and layout code, but no tests are modified. Please consider adding a test!

nox committed:
Return an error when animating `column-count: auto`

This changes the behaviour of animations of `column-count`
(hello Captain Obvious) but this is a bugfix.

emilio commented on components/style/values/generics/column.rs:
```diff
@@ -0,0 +1,30 @@
+    Integer(PositiveInteger),
+    /// The keyword `auto`.
+    #[animation(error)]
+    Auto,
```
Maybe mention in the commit message that you're changing the animation behavior for this, but that it's a bugfix?

nox replied:
Mention added.

nox commented:
@bors-servo r=emilio

bors-servo commented:
:pushpin: Commit 0a3f30f has been approved by `emilio`

<!-- @bors-servo r=emilio 0a3f30fb12d52d9a169dcd22798b88ad0a591030 -->

bors-servo commented:
:hourglass: Testing commit 0a3f30fb12d52d9a169dcd22798b88ad0a591030 with merge 5a3bd3955cb002c9f630f88bdfdc4ba124602bf5...
